pandas/tests/indexes/test_base.py

```diff
@@ -363,6 +364,19 @@ def test_constructor_dtypes_timedelta(self):
                         pd.TimedeltaIndex(list(values), dtype=dtype)]:
                 tm.assert_index_equal(res, idx)
 
+    def test_constructor_uint64(self):
```

Comment:
Reply to your comment about `pd.NaT` <a href="https://github.com/pandas-dev/pandas/pull/14937#discussion_r94510353">above</a>: the purpose is to see what the column dtype becomes when I set an element in the `UInt64Index` to a `NaN` value.  I added a comment to explain this.